utils/s3/client.cc

```diff
@@ -886,12 +886,22 @@ class client::readable_file : public file_impl {
     }
 
     virtual future<size_t> read_dma(uint64_t pos, void* buffer, size_t len, io_intent*) override {
+        co_await maybe_update_stats();
+        if (pos >= _stats->size) {
+            co_return 0;
+        }
+
         auto buf = co_await _client->get_object_contiguous(_object_name, range{ pos, len });
```

Comment:
We should adjust len so as not to overflow the size.

Reply:
S3 server seems to handle overflowing end of the range it correctly